shards/shards.go

```diff
@@ -580,98 +580,174 @@ func (ss *shardedSearcher) streamSearch(ctx context.Context, proc *process, q qu
 	shards, q = selectRepoSet(shards, q)
 	tr.LazyPrintf("after selectRepoSet shards:%d %s", len(shards), q)
 
-	var childCtx context.Context
+	if len(shards) == 0 {
+		return func() {}, nil
+	}
+
 	var cancel context.CancelFunc
 	if opts.MaxWallTime == 0 {
-		childCtx, cancel = context.WithCancel(ctx)
+		ctx, cancel = context.WithCancel(ctx)
 	} else {
-		childCtx, cancel = context.WithTimeout(ctx, opts.MaxWallTime)
+		ctx, cancel = context.WithTimeout(ctx, opts.MaxWallTime)
 	}
 
 	defer cancel()
 
-	mu := sync.Mutex{}
-	pendingPriorities := prioritySlice{}
-
-	g, ctx := errgroup.WithContext(childCtx)
-
-	// For each query, throttle the number of parallel
-	// actions. Since searching is mostly CPU bound, we limit the
-	// number of parallel searches. This reduces the peak working
-	// set, which hopefully stops https://cs.bazel.build from crashing
-	// when looking for the string "com".
-	//
-	// We do yield inside of the feeder. This means we could have num_workers +
-	// cap(feeder) searches run while yield blocks. However, doing it this way
-	// avoids needing to have synchronization in yield, so is done for
-	// simplicity.
-	feeder := make(chan *rankedShard, runtime.GOMAXPROCS(0))
-	g.Go(func() error {
-		defer close(feeder)
-		// Note: shards is sorted in order of descending priority.
-		for _, s := range shards {
-			// We let searchOneShard handle context errors.
-			_ = proc.Yield(ctx)
-			mu.Lock()
-			pendingPriorities.append(s.priority)
-			mu.Unlock()
-			feeder <- s
+	workers := runtime.NumCPU()
```

Comment:
need to use GOMAXPROCS due to cgroups and https://github.com/uber-go/automaxprocs
```suggestion
	workers := runtime.GOMAXPROCS(0)
```